lib/ipam/ipam_block_reader_writer_test.go

```diff
@@ -813,3 +951,70 @@ var _ = testutils.E2eDatastoreDescribe("IPAM affine block allocation tests", tes
 		})
 	})
 })
+
+var _ = testutils.E2eDatastoreDescribe("IPAM affine block allocation tests (kdd only)", testutils.DatastoreK8s, func(config apiconfig.CalicoAPIConfig) {
+
+	var (
+		bc  api.Client
+		net *cnet.IPNet
+		ctx context.Context
+	)
+
+	BeforeEach(func() {
+		var err error
+		bc, err = backend.NewClient(config)
+		Expect(err).NotTo(HaveOccurred())
+		bc.Clean()
+		ctx = context.Background()
+		_, net, err = cnet.ParseCIDR("10.0.0.0/26")
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should respect Kubernetes UID for affinities", func() {
```

Comment:
Should we have one for handles?

Reply:
We decided it wouldn't be necessary to include a race condition test for handles since we are using the same Delete flag approach for both resources so would follow that just testing affinities is sufficient to cover the race condition for both. If we were to change the approach for deleting handles, then it would be necessary to add in a separate test case; but for now, I think it's worth it to take out the technical loan for the time.

Reply:
@briansan this is just the UID tests, not the race condition tests.

I've added a test for handles respecting UID as well.